src/Core/Settings.h

```diff
@@ -587,7 +587,7 @@ class IColumn;
     M(Bool, allow_experimental_database_materialized_mysql, false, "Allow to create database with Engine=MaterializedMySQL(...).", 0) \
     M(Bool, allow_experimental_database_materialized_postgresql, false, "Allow to create database with Engine=MaterializedPostgreSQL(...).", 0) \
     M(Bool, system_events_show_zero_values, false, "When querying system.events or system.metrics tables, include all metrics, even with zero values.", 0) \
-    M(MySQLDataTypesSupport, mysql_datatypes_support_level, 0, "Which MySQL types should be converted to corresponding ClickHouse types (rather than being represented as String). Can be empty or any combination of 'decimal', 'datetime64', 'date2Date32' or 'date2String'. When empty MySQL's DECIMAL and DATETIME/TIMESTAMP with non-zero precision are seen as String on ClickHouse's side.", 0) \
+    M(MySQLDataTypesSupport, mysql_datatypes_support_level, {}, "Which MySQL types should be converted to corresponding ClickHouse types (rather than being represented as String). Can be empty or any combination of 'decimal', 'datetime64', 'date2Date32' or 'date2String'. When empty MySQL's DECIMAL and DATETIME/TIMESTAMP with non-zero precision are seen as String on ClickHouse's side.", 0) \
```

Comment:
Since this setting is a MultiEnum, `0` was passed to this constructor indicating an empty `MultiEnum`: https://github.com/ClickHouse/ClickHouse/blob/335e7bf7ef5ab3ed820caaed7c7aec1caad4096c/src/Core/SettingsFields.h#L470

I've modified the code to store a vector instead of a `MultiEnum` in `SettingFieldMultiEnum` and removed this constructor. To maintain the same behavior, we should initialize the setting with an empty value. After the change, we have these constructors: https://github.com/ClickHouse/ClickHouse/blob/d336b60a1943462549b5385f40d4efdf7ac8fafa/src/Core/SettingsFields.h#L467-L469

Let's specify `std::vector<MySQLDataTypesSupport>{}` here instead of `{}` to make it clearer what value the setting will have.

Reply:
Let's do `using MySQLDataTypesSupportList = std::vector<MySQLDataTypesSupport>;` and use `MySQLDataTypesSupportList{}` everywhere.

Reply:
Ok, then it probably makes sense to introduce `*List` using's for all multi enum settings (now only `JoinAlgorithm` and `MySQLDataTypesSupport`) so I added this definition into `DECLARE_SETTING_MULTI_ENUM`